compiler/rustc_resolve/src/imports.rs

```diff
@@ -1348,27 +1355,26 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     target_bindings[ns].get(),
                 ) {
                     Ok(other_binding) => {
-                        is_redundant[ns] = Some(
-                            binding.res() == other_binding.res() && !other_binding.is_ambiguity(),
-                        );
+                        if_redundant =
+                            binding.res() == other_binding.res() && !other_binding.is_ambiguity();
+                        is_redundant[ns] = Some(if_redundant);
                         redundant_span[ns] = Some((other_binding.span, other_binding.is_import()));
                     }
-                    Err(_) => is_redundant[ns] = Some(false),
+                    Err(_) => if_redundant = false,
```

Comment:
This is a change in behavior, and not a correct one, I think.
`is_redundant[ns] = Some(if_redundant);` was previously executed in this case too.

Reply:
Thank you.
I think the requirement of `!is_redundant.is_empty() && is_redundant.present_items().all(|is_redundant| is_redundant)` 
 is that PerNS {value ns: None, type ns: None, macro ns: None} must have at least one value that is Some (true) and no value that is Some (false). I think it's possible to set `is_redundant` to 'bool' and use `is_redundant && !redundant_span.is_empty()` to replace, `is_ redundant` guarantees that there is no Some (false), `!redundant_span.is_empty()` ensures that there is at least one Some (true)